lucene/analysis/nori/src/tools/java/org/apache/lucene/analysis/ko/util/BinaryDictionaryWriter.java

```diff
@@ -137,14 +139,17 @@ public int put(String[] entry) {
       flags |= BinaryDictionary.HAS_READING;
     }
 
-    assert leftId < 8192; // there are still unused bits
-    assert posType.ordinal() < 4;
+    if (leftId >= ID_LIMIT) {
+      throw new IllegalArgumentException("leftId >= " + ID_LIMIT + ": " + leftId);
+    }
+    if (posType.ordinal() >= 4) {
+      throw new IllegalArgumentException("posType.ordinal() >= " + 4 + ": " + posType.ordinal());
```

Comment:
Maybe also log posType itself? If this ever happens, that should help

Reply:
+1.

Which log style is better?
1) `throw new IllegalArgumentException("posType.ordinal() >= " + 4 + ": " + posType.name());`
2) `throw new IllegalArgumentException("posType should be MORPHOME or COMPOUND or INFLECT or PREANALYSIS" + ": " + posType.name());`

Reply:
(1), please!

Reply:
Thanks :)